lang/py/src/avro/schema.py

```diff
@@ -405,6 +420,106 @@ def __eq__(self, that):
     return self.props == that.props
 
 #
+# Decimal Type
+#
+class DecimalSchema(LogicalSchema):
+  def __init__(self, decimal_type, precision, scale):
```

Comment:
To better match the spec, we should make the `scale` optional and default to `0` (here and in the subclasses). `decimal_type` might be a bit confusing too, maybe `underlying_type` or just `type` would be more explicit.